docs/logs/getting-started-aspnetcore/README.md

```diff
@@ -0,0 +1,150 @@
+telemetry.sdk.name: opentelemetry
+telemetry.sdk.language: dotnet
+telemetry.sdk.version: 1.6.0-rc.1.24
+
+LogRecord.Timestamp:               2023-09-06T23:00:46.1639248Z
+LogRecord.TraceId:                 3507087d60ae4b1d2f10e68f4e40784a
+LogRecord.SpanId:                  c51be9f19c598b69
+LogRecord.TraceFlags:              None
+LogRecord.CategoryName:            Program
+LogRecord.Severity:                Info
+LogRecord.SeverityText:            Information
+LogRecord.Body:                    Food `{name}` price changed to `{price}`.
+LogRecord.Attributes (Key:Value):
+    name: artichoke
+    price: 9.99
+    OriginalFormat (a.k.a Body): Food `{name}` price changed to `{price}`.
+LogRecord.EventId:                 2
+LogRecord.EventName:               FoodPriceChanged
+LogRecord.ScopeValues (Key:Value):
+[Scope.0]:SpanId: c51be9f19c598b69
+[Scope.0]:TraceId: 3507087d60ae4b1d2f10e68f4e40784a
+[Scope.0]:ParentId: 0000000000000000
+[Scope.1]:ConnectionId: 0HMTF5KCP0BU9
+[Scope.2]:RequestId: 0HMTF5KCP0BU9:00000003
+[Scope.2]:RequestPath: /
+
+Resource associated with LogRecord:
+service.name: getting-started-aspnetcore
+service.instance.id: d62727d2-3e80-4780-a984-e85fc861e321
+telemetry.sdk.name: opentelemetry
+telemetry.sdk.language: dotnet
+telemetry.sdk.version: 1.6.0-rc.1.24
+```
+
+Congratulations! You are now collecting logs using OpenTelemetry.
+
+What does the above program do?
+
+The program has added OpenTelemetry as a [logging
+provider](https://learn.microsoft.com/dotnet/core/extensions/logging-providers)
+to the existing logging pipeline. OpenTelemetry SDK is then configured with a
+`ConsoleExporter` to export the logs to the console. In addition,
+`OpenTelemetryLoggerOptions.IncludeScopes` is enabled so the logs will include
+the [log
+scopes](https://learn.microsoft.com/dotnet/core/extensions/logging#log-scopes).
+From the console output we can see the log scopes that are coming from the
+ASP.NET Core framework, and we can see logs from both our logger and the ASP.NET
+Core framework loggers.
+
+The example has demonstrated the best practice from ASP.NET Core by injecting
+generic `ILogger<T>`:
+
+```csharp
+app.MapGet("/", (ILogger<Program> logger) =>
+{
+    logger.FoodPriceChanged("artichoke", 9.99);
+
+    return "Hello from OpenTelemetry Logs!";
+});
+```
+
+To achieve structured logging and high performance, [compile-time logging source
+generation](https://docs.microsoft.com/dotnet/core/extensions/logger-message-generator)
+is used:
+
+```csharp
+public static partial class ApplicationLogs
+{
+    [LoggerMessage(EventId = 1, Level = LogLevel.Information, Message = "Starting the app...")]
+    public static partial void StartingApp(this ILogger logger);
+
+    [LoggerMessage(EventId = 2, Level = LogLevel.Information, Message = "Food `{name}` price changed to `{price}`.")]
+    public static partial void FoodPriceChanged(this ILogger logger, string name, double price);
+}
+```
+
```

Comment:
I think it would be fine to mention somewhere around here that there are also less verbose alternatives like logger.LogInformation(...) but you lose out on some performance and type checked parameters if you choose to use them. Personally I'd also be fine not mentioning it, but I think both Sam and Fowler were in favor of covering the option. I'm happy as long as anyone whose attention expired before reaching this point is going to use LoggerMessage, which I consider the option with higher likelihood of a good outcome in the long run.

Reply:
We should show the inverse. Show log information and then mention logger message.

Reply:
Those are better covered in ILogger tutorial/docs. This doc is intended to show how to use OpenTelemetry with ILogger, so its best to just show *the recommended*  way of logging, instead of showing all/more-than-one possible ways of ILogger usage itself.

Reply:
> We should show the inverse. Show log information and then mention logger message.

As I understand it, @reyang's team has concrete past experience that if we do that it leads to questions and support tickets. A subset of devs copy whatever code is in the first snippet, don't read the following text, then scale up that usage. Its a deliberate choice that if we only have their attention span long enough to copy/paste one thing, lets have that thing be the performant and scalable option rather than the succinct option.

Reply:
> > We should show the inverse. Show log information and then mention logger message.
> 
> As I understand it, @reyang's team has concrete past experience that if we do that it leads to questions and support tickets. A subset of devs copy whatever code is in the first snippet, don't read the following text, then scale up that usage.

+100, based on the support tickets and questions we've observed during the past 2 years, I think we should only provide pit of success, and we must not show any negative example that are not recommended for real production development. In addition, with the trend of LLM/ChatGPT/CoPilot, I think we should consider how good/bad examples could influence AI.

Reply:
> we must not show any negative example that are not recommended for real production development

I wouldn't go as far as saying the other APIs are unsuitable or not recommended for production, they just have a suitable scale threshold that is lower than LoggerMessage.

Reply:
> > we must not show any negative example that are not recommended for real production development
> 
> I wouldn't go as far as saying the other APIs are unsuitable or not recommended for production, they just have a suitable scale threshold that is lower than LoggerMessage.

Fair, I think we should also try to avoid slippery slopes.